### PR TITLE
Update Changelog for 1.8.7 release

### PR DESCRIPTION
**Pull request by amartinezfayo**
Update Changelog for 1.8.7 release.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,19 @@
 # Changelog
 
+## [1.8.7] - 2023-12-21
+
+### Added
+
+- Agents can now be configured with an availability target, which establishes the minimum amount of time desired to gracefully handle server or agent downtime, influencing how aggressively X509-SVIDs should be rotated (#4599)
+- SyncAuthorizedEntries RPC, which allows agents to only sync down changes instead of the entire set of entries. Agents can be configured to use this new RPC through the `use_sync_authorized_entries` experimental setting (#4648)
+- Experimental support to allow using an events based cache, to reduce overhead on the database (#4379, #4411, #4527, #4451, #4562, #4723, #4731)
```
Comment (azdagron) [inline CHANGELOG.md:9]: nit:
```suggestion
- Experimental support for an events based entry cache which reduces overhead on the database (#4379, #4411, #4527, #4451, #4562, #4723, #4731)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,19 @@
...
+
+### Changed
+
+- The maximum number of open database connections in the datastore now defaults to 100 instead of unlimited (#4656)
+- Agents now shut down when can't synchronize entries with the server because it returned an unknown authority error (#4617)
```
Comment (azdagron) [inline CHANGELOG.md:14]:
```suggestion
- Agents now shut down when they can't synchronize entries with the server due to an unknown authority error (#4617)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,19 @@
...
+- Agents no longer use legacy files for storing bundle and agent SVID information, and legacy files are removed (#4717)
```
Comment (azdagron) [inline CHANGELOG.md:15]: I don't know that I'd include this one... if so, maybe stick it under `Removed` and maybe word it like:

```
- Agents no longer maintains agent SVID and bundle information in the legacy paths in the data directory (#4717)
- ```

**Review by azdagron — COMMENTED** [review]
Thanks so much @amartinezfayo ! Just a few comments.

File: CHANGELOG.md (hunk 3 above)
Comment (amartinezfayo) [reply CHANGELOG.md:15]: I think it's worth including it, I'll move it to the Removed section with the suggested wording. Thanks!

**Review by azdagron — APPROVED** [review]

**Review by MarcosDY — APPROVED** [review]

**Pull request merged**